### PR TITLE
CLOUD-87652 Added local-dev install script.

### PR DESCRIPTION
**Pull request by mvicianHW**

### CODE REVIEW

File: install-dev [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+main() {
+    declare branch=${1:-master}
+
+    local os=$(uname -s | tr "[A-Z]" "[a-z]")
+    local latestartifacts="project/github/$GITHUB_USERNAME/$GITHUB_PROJECT/latest/artifacts?branch=$branch&filter=successful"
+    local artifacts=$(circle $latestartifacts | tr -d '[:space:]')
+    local message=$(echo $artifacts | jq -c ".message" 2>/dev/null)
+   
+    if [ "$?" -gt 0 ]; then 
+        echo "No artifact found for branch: 'master' at https://circleci.com/gh/$GITHUB_USERNAME/$GITHUB_PROJECT"
```
Comment (mhmxs) [inline install-dev]: Branch is a local variable would be nice to use it here.

File: install-dev [hunk 2]
```diff
@@ -0,0 +1,53 @@
+: ${GITHUB_USERNAME:=hortonworks}
```
Comment (mhmxs) [inline install-dev]: Missing shebang

File: install-dev [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    local os=$(uname -s | tr "[A-Z]" "[a-z]")
+    local latestartifacts="project/github/$GITHUB_USERNAME/$GITHUB_PROJECT/latest/artifacts?branch=$branch&filter=successful"
+    local artifacts=$(circle $latestartifacts | tr -d '[:space:]')
+    local message=$(echo $artifacts | jq -c ".message" 2>/dev/null)
+   
+    if [ "$?" -gt 0 ]; then 
+        echo "No artifact found for branch: 'master' at https://circleci.com/gh/$GITHUB_USERNAME/$GITHUB_PROJECT"
+        exit 1
+    elif [ "${message}" != "" ]; then
```
Comment (mhmxs) [inline install-dev]: We prefer bash style elif [[ ! "${message}" ]] or [[ -n "${message}" ]]

File: install-dev [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        "$url" "$@"
+}
+
+main() {
+    declare branch=${1:-master}
+
+    local os=$(uname -s | tr "[A-Z]" "[a-z]")
+    local latestartifacts="project/github/$GITHUB_USERNAME/$GITHUB_PROJECT/latest/artifacts?branch=$branch&filter=successful"
+    local artifacts=$(circle $latestartifacts | tr -d '[:space:]')
+    local message=$(echo $artifacts | jq -c ".message" 2>/dev/null)
```
Comment (mhmxs) [inline install-dev]: `jq` is a third party tool, most of the operating systems doesn't contain by default. .deps folder contains `jq` under bin, but requires to initialize `cbd` first.

File: install-dev [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        echo "No artifact found for branch: 'master' at https://circleci.com/gh/$GITHUB_USERNAME/$GITHUB_PROJECT"
+        exit 1
+    elif [ "${message}" != "" ]; then
+        echo "CircleCI API returned with message: $message"
+        echo "API call: $circleciapi/$latestartifacts"
+        exit 1
+    fi
+
+    local url=$(echo $artifacts | jq -c ".[].url" | grep $os | tr -d '"')
+    if [ "$?" -gt 0 ]; then
```
Comment (mhmxs) [inline install-dev]: Quotas not needed, because we compare two numbers.

File: install-dev [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        exit 1
+    fi
+
+    local url=$(echo $artifacts | jq -c ".[].url" | grep $os | tr -d '"')
+    if [ "$?" -gt 0 ]; then 
+        echo "No artifacts found for os: $os at https://circleci.com/gh/$GITHUB_USERNAME/$GITHUB_PROJECT"
+        exit 1
+    fi
+
+    local dest=/bin
```
Comment (mhmxs) [inline install-dev]: /bin contains "system" binaries, so /usr/bin would be better and match more to POSIX standards. "/bin Essential command binaries that need to be available in single user mode; for all users, e.g., cat, ls, cp."

**Comment by mvicianHW** [comment]
Updated. Please check, thanks.

**Pull request merged**